### DISCUSSION TITLE
Does toolkit method "select" works with SVG and MusicXML?

### DISCUSSION THREAD

**Discussion by andre2007** [rism-digital/verovio#3769] [Q&A] [answered]
Hi,
I call the C wrapper functions of Verovio from D and have some issues with the `select` function.

```d
auto toolkit = new VrvToolkit("data");
writeln(toolkit.getVersion()); // 4.2.0-dev[undefined]
toolkit.loadFile("complex.musicxml");
toolkit.select(`{"measureRange":"3-5"}`);
toolkit.renderToSVGFile("complex.svg");
```

Although I specify that measure `3-5` should be rendered, the whole page is rendered.
One observation: if I pass some invalid JSON to the select method, this is detected by
Verovio, that means the value is passed fine, but it seems it is not evaluted.

Did I something wrong or does the select method not work together with MusicXML and SVG?

Kind regards
André

[complex.zip](https://github.com/user-attachments/files/16921641/complex.zip)

**Comment by andre2007** [comment]
I just tried calling `redoLayout` before method call `renderToSVGFile` this causes an exception:

![image](https://github.com/user-attachments/assets/d358a5ff-773a-41cd-95d5-d9a2ae337621)

**Reply by lpugin** [reply, marked as answer]
You need to do `toolkit.select({"measureRange": "1-3"})`

**Comment by andre2007** [comment]
Unfortunately it does not work. It causes an exception with the sample musicxml attached, if I call `redoLayout` after the `select` method call. Without calling `redoLayout`, the whole page is renderted.

(In the C wrapper I pass the JSON as string, beside that, the coding line you proposed is identical).

Kind regards
André

**Reply by lpugin** [reply]
The code I posted was in Python and you are right it needs to be passed as a string in C/C++. It works for me both in Python and C++ using Verovio as a library, so it must be related to your binding or to Windows.

![image](https://github.com/user-attachments/assets/f7be9c46-ca72-41fc-b910-7124093fabfb)

**Comment by andre2007** [comment]
Thanks. Issue was caused by the build instructions. With these build instructions, it seems to work fine
```
cd cmake
cmake -DBUILD_AS_LIBRARY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=TRUE -B build 
cmake --build build --config Release
```

**Reply by lpugin** [reply]
Could you check if changing to `--config Debug` works too? If not, then this means an `assert` might fail and this should be investigated.

**Reply by lpugin** [reply]
I just check and there is a problem with an `assert`.
